### PR TITLE
Allow overriding debug function.

### PR DESCRIPTION
**Pull request by pmcnr-hx**
Add a function to the `debugging` module that allows overriding the debugging functions for each of the namespaces with a user provided function.

* [x] :heart: 
* [x] :two_hearts:
* [x] :heart: x :three:

### CODE REVIEW

File: documentation/debugging.md [hunk 1]
```diff
@@ -25,3 +25,7 @@ $ DEBUG=jsonApi:handler:find npm test
 ```
 $ DEBUG=jsonApi:handler:* npm test
 ```
+
+### Integration with application logging
+
+If you wish to integrate `jsonapi-server` debug output with your application's logging solution, you can override the functions that are invoked for generating the output of each of the supported namespaces by invoking the package's `debugging.__setLoggers(loggerFactory)` function where the `loggerFactory` parameter is a function that expects a namespace string parameter and returns a logger function for the namespace.
```
Comment (theninj4) [inline documentation/debugging.md]: Could you put an example beneath this? It's not entirely obvious by reading this paragraph and the code is both elegant and not immediately obvious.

In other words - people will ask us a lot of questions about how this works, so lets just add an example we can refer them back to.

File: lib/debugging.js [hunk 2]
```diff
@@ -1,18 +1,33 @@
...
-  errors: require("debug")("jsonApi:errors"),
-  requestCounter: require("debug")("jsonApi:requestCounter")
+  include: debug("jsonApi:include"),
+  filter: debug("jsonApi:filter"),
+  validationInput: debug("jsonApi:validation:input"),
+  validationOutput: debug("jsonApi:validation:output"),
+  validationError: debug("jsonApi:validation:error"),
+  errors: debug("jsonApi:errors"),
+  requestCounter: debug("jsonApi:requestCounter"),
+  __setLoggers: setLoggersHelper.bind(null, debugging)
```
Comment (theninj4) [inline lib/debugging.js]: Is `setLoggers` a good choice of function name? Should it be more like `.__redirectOutputTo(winston)` or similar?

Comment (pmcnr-hx) [inline lib/debugging.js]: Agreed. I was struggling to come up with a decent name. I'll rename accordingly.

**Comment by theninj4** [comment]
Works a treat :+1:

**Comment by reecemillsom** [comment]
taking 3️⃣

**Comment by DuncanFenning** [comment]
got :two_hearts:

**Comment by reecemillsom** [comment]
👍

**Comment by DuncanFenning** [comment]
go for it 👍

**Pull request merged**